### DISCUSSION TITLE
macOS CI with Anka, Intel Mac mini: perf benefit of 64GB vs. 32 GB?

### DISCUSSION THREAD

**Discussion by claybridges** [MobileNativeFoundation/discussions#78] [General]
Our team will be prototyping a self-managed CI replacement for Buddybuild, for building and testing iOS apps and frameworks. We'll likely be using Anka on MacStadium. Does anybody using Anka in a similar context have experience indicating performance benefits to having 64GB (vs. 32GB) of memory? I'm specifically interested in the effect on UI tests on the iOS Simulator (death to flakey tests, _always!_)

Since there will be two VMs per physical machine, the VMs would be getting bumped from ~16GB to ~32GB.

**Comment by keith** [comment]
Some related conversations:

- https://github.com/MobileNativeFoundation/discussions/discussions/68
- https://github.com/MobileNativeFoundation/discussions/discussions/63
- https://github.com/MobileNativeFoundation/discussions/discussions/23

**Comment by JoeSSS** [comment]
I have quite some experience with this :) For two VMs, 32 gb is enough. 2 VMs 14 gb each + 4gb should be left for the system itself. We spawn 3 VMs on 32gb sometimes and it still works fine.

**Comment by JoeSSS** [comment]
I would suggest you measure the RAM usage during building/testing instead. For example, for us unit testing takes 10gb of ram, we give 9gb to the system so 1 gb goes to swap, but this is still an acceptable slow down for gathering additional executor. Overall, under pressure, our CI can be blocked because of a lack of executors not because the building is too slow.

**Comment by CognitiveDisson** [comment]
I would say it highly depends on the project and use case. I came to the conclusion that 32GB is enough - Mac mini 6 cores/32GB RAM/512GB SSD,  medium-sized project (if that says something), 2 VMs per node, build and running tests on one simulator.
For medium sized projects, 16GB is not enough even for one VM. And it is noticeable that sometimes the machine starts using swap, which is not good.
Optimal performance for Mac mini was with two VMs. With a larger number, there was a problem with the disk I/O and the CPU.
It also works with three, but there are already drawdowns.
